### PR TITLE
[Train/Tune] Warn pending deprecation for `ray.train.Trainer` and `ray.tune` DistributedTrainableCreators

### PR DESCRIPTION
**Pull request by amogkam**
`ray.train.Trainer` and `ray.tune.integration.*.DistributedTrainableCreator` will be deprecated in Ray 2.0 in favor of Ray AIR. In Ray 1.13, we should warn about this pending deprecation.

First step towards https://github.com/ray-project/ray/issues/23014

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by Yard1** [comment]
Add `@Deprecated` decorators as well?

**Comment by amogkam** [comment]
Ah @Yard1 we will officially denote as deprecated for Ray 2.0. But for Ray 1.13, we just want to print a warning message saying that this will happen in the future.

**Comment by Yard1** [comment]
The decorator only adds a message to the doc that this API will be removed in a future ray releasw

**Comment by amogkam** [comment]
The decorator official denotes the APIs as deprecated (even though may actually be removed in a later release). We don't want to deprecate these APIs in Ray 1.x, and will wait for the major release to do so.

**Comment by amogkam** [comment]
Basically we don't want this part yet: `DEPRECATED: This API is deprecated`

**Comment by Yard1** [comment]
I see, ok! Seems odd to have this warning on init but not in the docstring somewhere, though. We want to dissuade people from using those in the first place, right? Having a mention in the docstring about the pending depreciation would steer users who land on the docs page towards Train.

**Comment by amogkam** [comment]
Yeah that's a good point. I can definitely manually add it, but I'm not sure if we have standardized on any particular policy across all of Ray.

For reference, this is what Python does:

> Python library uses [warnings.warn()](https://docs.python.org/3/library/warnings.html#warnings.warn) to emit a deprecation warning in the body of functions.
3 categories are used: [DeprecationWarning](https://docs.python.org/3/library/exceptions.html#DeprecationWarning), [PendingDeprecationWarning](https://docs.python.org/3/library/exceptions.html#PendingDeprecationWarning) and [FutureWarning](https://docs.python.org/3/library/exceptions.html#FutureWarning).
The docstring doesn’t show anything about deprecation.
The documentation warns about some, but not all, deprecated usages.

From https://deprecated.readthedocs.io/en/latest/white_paper.html.

**Comment by xwjiang2010** [comment]
How about others under ray/tune/integration?

**Comment by amogkam** [comment]
@xwjiang2010 these are the only ones we want to deprecate right? (These are the only ones with DistributedTrainableCreators).

**Comment by xwjiang2010** [comment]
@amogkam Ah I see. That makes sense.
One more question, how about the docs? Like should we update examples [here](DistributedTrainableCreator)? Or maybe we will wait for 2.0.0 when it's truly deprecated?

**Review by xwjiang2010 — APPROVED** [review]

**Review by matthewdeng — APPROVED** [review]

**Review by Yard1 — APPROVED** [review]

**Comment by amogkam** [comment]
@xwjiang2010 yep let's remove the docs once they are fully deprecated!

File: python/ray/train/trainer.py [hunk 1]
```diff
@@ -135,6 +136,16 @@ def __init__(
         logdir: Optional[str] = None,
         max_retries: int = 3,
     ):
+        warnings.warn(
+            "The `ray.train.Trainer` API will be deprecated in Ray "
+            "2.0, and will be replaced by Ray AI Runtime (Ray AIR). Ray AIR ("
+            "https://docs.ray.io/en/master/ray-air/getting-started.html) will "
```
Comment (bveeramani) [inline python/ray/train/trainer.py:142]: Should we point users to latest versions of the docs (_i.e_, https://docs.ray.io/en/latest) as opposed to master?

**Review by bveeramani — CHANGES_REQUESTED** [review]
Had a question about the documentation linked. Otherwise LGTM

File: python/ray/train/trainer.py (hunk 1 above)
Comment (amogkam) [reply python/ray/train/trainer.py]: Updated!

**Review by bveeramani — APPROVED** [review]
LGTM

**Pull request merged**